src/ui/component/dateInput/dateInput.tsx

```diff
@@ -0,0 +1,51 @@
+import { useMemo, type FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Icon } from '@/ui/component/icon/icon'
+import './dateInput.scss'
+
+type DateInputProps = {
+  date?: Date
+}
+
+export const DateInput: FC<DateInputProps> = ({ date }) => {
+  const { i18n } = useTranslation()
+  const locale = i18n.language
+
+  const localizedDateFormatter = useMemo(
+    () => new Intl.DateTimeFormat(locale, { year: 'numeric', month: '2-digit', day: '2-digit' }),
+    [locale]
+  )
+
+  const localizedDatePattern = useMemo(
+    () =>
+      localizedDateFormatter
+        .formatToParts(date)
+        .map(({ type, value }) => {
+          switch (type) {
+            case 'day':
+              return 'DD'
+            case 'month':
+              return 'MM'
+            case 'year':
+              return 'YY'
+            default:
+              return value
+          }
+        })
+        .join(''),
+    [localizedDateFormatter, date]
+  )
+
```

Comment:
Utils functions should be implemented in the dedicated `util` folder and be tested accordingly

Reply:
Functions have been reworked in [7c49d90](https://github.com/okp4/dataverse-portal/pull/460/commits/7c49d909dac135b9acd3654f79891350f3fc2b1f) and tested in [43b9c1e](https://github.com/okp4/dataverse-portal/pull/460/commits/43b9c1ebee40631c343e1f6b3d3606e2491ae0b2).

Thanks for the feedback, it helped to improve function signature and behavior thanks to testing.